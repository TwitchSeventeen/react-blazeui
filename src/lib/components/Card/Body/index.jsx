import React from 'react';
import PropTypes from 'prop-types';

function Body(props) {
  const { children, ...other } = props;
  return (
    <div {...other} className="c-card__body">
      { children }
    </div>
  );
}

Body.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Body.defaultProps = {
  children: null,
};

export default Body;
