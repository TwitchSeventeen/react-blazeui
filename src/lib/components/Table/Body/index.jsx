import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Body(props) {
  const {
    children, className, ...other
  } = props;

  const classes = classNames(
    'c-table__body',
    className,
  );

  return (
    <tbody className={classes} {...other}>
      { children }
    </tbody>
  );
}

Body.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
};

Body.defaultProps = {
  className: null,
  children: null,
};

export default Body;
