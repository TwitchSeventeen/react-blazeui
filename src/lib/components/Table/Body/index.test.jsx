import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Body from './index';

describe('Body', () => {
  it('should be defined', () => {
    expect(Body).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Body>
        lorem ipsum dolor
      </Body>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
