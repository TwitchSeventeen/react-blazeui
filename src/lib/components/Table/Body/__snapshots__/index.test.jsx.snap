// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders correctly 1`] = `
<tbody
  className="c-table__body"
>
  lorem ipsum dolor
</tbody>
`;
